File: Auth/Auth.Infra/RefreshToken/InMemory/InMemoryRefreshTokenRepository.cs
using Auth.App.Interface.RefreshToken;

namespace Auth.Infra.RefreshToken.InMemory
{
    internal class InMemoryRefreshTokenRepository : IRefreshTokenRepository
    {
        private readonly Dictionary<string, RefreshTokenRecord> _refreshTokens = new();

        public Task UpdateAsync(RefreshTokenRecord record)
        {
            _refreshTokens[record.Token] = record;
            return Task.CompletedTask;
        }

        public Task<RefreshTokenRecord?> GetByTokenAsync(string token)
        {
            _refreshTokens.TryGetValue(token, out var record);
            return Task.FromResult(record);
        }

        public Task DeleteAsync(string token)
        {
            _refreshTokens.Remove(token);
            return Task.CompletedTask;
        }

        public void Migrate()
        {
            
        }

        public Task<RefreshTokenRecord?> GetAsync(string token)
        {
            _refreshTokens.TryGetValue(token, out var record);
            return Task.FromResult(record);
        }

        public Task CreateAsync(RefreshTokenRecord record)
        {
            _refreshTokens[record.Token] = record;
            return Task.CompletedTask;
        }

        public Task DeleteAsync(Guid userId)
        {
            var record = _refreshTokens.FirstOrDefault(kvp => kvp.Value.UserId == userId);
            if (record.Key != null)
            {
                _refreshTokens.Remove(record.Key);
            }
            return Task.CompletedTask;
        }
    }
}

File: NodeAuth/src/repos/inMemory/InMemoryRefreshTokenRepository.ts
import { RefreshTokenRecord } from "../../models/RefreshTokenRecord";
import { IRefreshTokenRepository } from "../interfaces/IRefreshTokenRepository";

export class InMemoryRefreshTokenRepository implements IRefreshTokenRepository {
  private readonly refreshTokens = new Map<string, RefreshTokenRecord>();

  async getAsync(token: string): Promise<RefreshTokenRecord | null> {
    return this.refreshTokens.get(token) ?? null;
  }

  async createAsync(record: RefreshTokenRecord): Promise<void> {
    this.refreshTokens.set(record.token, record);
  }

  async updateAsync(record: RefreshTokenRecord): Promise<void> {
    this.refreshTokens.set(record.token, record);
  }

  async deleteAsync(token: string): Promise<void> {
    this.refreshTokens.delete(token);
  }

  async deleteByUserIdAsync(userId: string): Promise<void> {
    for (const [token, record] of this.refreshTokens.entries()) {
      if (record.userId === userId) {
        this.refreshTokens.delete(token);
        return;
      }
    }
  }

  migrate(): void {
    // No-op for in-memory storage
  }
}
